### ISSUE TITLE
MA0042 is reported for completed tasks

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#564]
### Version of the Meziantou.Analyzer NuGet package

2.0.68

### Rule Identifier

MA0042

### Target Framework

net8.0

### C# Language version

C# 12

### Description

MA0042 is reported after Task.WhenAll

### Reproduction Steps

```csharp
var task1 = Task.FromResult(42);
var task2 = Task.FromResult("hello");
await Task.WhenAll(task1, task2);

_ = task1.Result; // <-- MA0042 is reported here while task1 is definitely completed
```

**Comment by meziantou** [comment]
This is something that is not supported. It would require non-trivial work to do a working data flow analysis. As the pattern is not common, I suggest disabling the warning using an attribute or a `#pragma warning disable`.

For this specific pattern, I also suggest using a simpler way to do a `Task.WhenAll` followed by `.Result` https://www.meziantou.net/get-the-result-of-multiple-tasks-in-a-valuetuple-and-whenall.htm
